### ISSUE TITLE
vlc.MediaPlayer() instances are causing memory leaks

### ISSUE THREAD

**Issue by parklez** [parklez/twitch-bot#1]
Whenever a vlc.MediaPlayer is initialized and played, the object frees the memory from the audio but not from itself once it's done playing, so over time, memory is being eaten. For normal situations this won't be too impactful.

**Comment by parklez** [comment]
Python's del method isn't helping, even after calling the garbage collector:
```python
import vlc
import gtts

sound = gtts.gTTS('heyoooooooo')
a = [vlc.MediaPlayer(sound.get_urls()[0]) for
     item in range(0, 400)]

def print_mem():
    mem_total = 0
    for item in a:
        mem_total += item.__sizeof__()
    print(f'Total Mem used by {len(a)} items:', mem_total)
    input()

print_mem()

for item in a:
    del item

print_mem()

import gc
gc.collect()

print_mem()
```
Another approach would be to dynamically use 10 or so pre-allocated MediaPlayer instances that can be used on demand, but will see about that in the future. VLC is doing great work dealing with media.

**Comment by parklez** [comment]
Decided to research further, apparently there's a release() method to terminate a player's instance. This will be useful for dynamic sounds in the bot. For the future "music player" this won't be necessary.
https://www.geeksforgeeks.org/python-vlc-mediaplayer-releasing-object/?ref=rp

**Comment by parklez** [comment]
Threads can be a cheap way to fix this, I will consider other (better?) possibilities for concurrency.

```python
"""
DON'T TRY THIS AT HOME!

This code is meant to test how much memory will be consumed over time
after creating, playing and releasing vlc.MediaPlayer() instances.

Result: After 400 threads, memory stayed around 40mb which is acceptable.
"""

import vlc
import gtts
import threading
import time


def play(audio):
    audio.play()
    running = True
    while audio.get_state() != vlc.State.Ended:
        pass
    audio.release()

request = gtts.gTTS('heyoooooooo')

a = [vlc.MediaPlayer(request.get_urls()[0]) for
     item in range(0, 400)]

b = [threading.Thread(target=play, args=(x,)) for x in a]

for item in b:
    item.start()
    time.sleep(0.01)

del(b)
del(a)
del(item)
input()
```

**Comment by parklez** [comment]
Solved with [532be0c](https://github.com/parklez/twitch-bot/commit/532be0c51132abb51165db990703743d53df14a7) for now.
